napari/_vispy/layers/labels.py

```diff
@@ -68,8 +73,13 @@
     // we get a different value:
     // - if it's the empty key, exit;
     // - otherwise, it's a hash collision: continue searching
-    while ((abs(found - t) > 1e-8) && (abs(found - empty) > 1e-8)) {
+    float initial_t = t;
+    float lut_size = LUT_shape.x * LUT_shape.y
```

Comment:
There is one problem here. As float has precision up to 2**23. Should I limit the number of labels to 2**23 in function to construct?

Reply:
I would say so!

Reply:
Actually, should we also change the inequality comparisons from using `1e-8` to `1e23`? Or is that unrelated?

Reply:
> Should I limit the number of labels to 2**23 in function to construct?

What? No! The issue with the float32 2**23 is that labels *might* end up with the same color. But they might not! 2**25 + 100 and 2**25 + 1000 will fall in different bins:

```
In [21]: np.float32(2**25 + 1000) - np.float32(2**25 + 100)
Out[21]: 900.0
```

Occasional key overwriting is better than crashing! And this issue is already tracked by #6084.

Reply:
I mean that limit option to provide more than `2**23` different labels.

Reply:
Yeah the question is about total number of labels... though to be fair, I guess even then it's better to just have guaranteed duplicates than simply crashing :shrug:

Reply:
Even for total number it doesn't make sense? labels `1` to `2**23` will all work fine together. Then you have every other label between `2**23` and `2**24`. And so on. So there are many more than ~8M labels that could be separately rendered.

Reply:
I have updated the limitations. Could you check them? 
I also need to make PR to vispy (for future improvements)

Reply:
> Even for total number it doesn't make sense? labels `1` to `2**23` will all work fine together. Then you have every other label between `2**23` and `2**24`. And so on. So there are many more than ~8M labels that could be separately rendered.

The limitation comes from the condition stop of the while loop. As I change to use int (`int32`) instead of float the problem is solved in another way.

Reply:
> The limitation comes from the condition stop of the while loop. As I change to use int (int32) instead of float the problem is solved in another way.

OOOOooooooooOOOOH! I get it now! 😅 Clever. 😅